network.go

```diff
@@ -957,10 +989,36 @@ func (n *network) delete(force bool) error {
 		return &UnknownNetworkError{name: name, id: id}
 	}
 
-	if len(n.loadBalancerIP) != 0 {
-		endpoints := n.Endpoints()
-		if force || (len(endpoints) == 1 && !n.ingress) {
-			n.deleteLoadBalancerSandbox()
+	// only remove ingress on force removal or explicit ingress removal
+	if n.ingress && !force && !delIngress {
+		return &ActiveEndpointsError{name: n.name, id: n.id}
+	}
+
+	// Check that the network is empty
+	var emptyCount uint64
+	if n.hasLoadBalancerEndpoint() {
+		emptyCount = 1
+	}
+	if !force && n.getEpCnt().EndpointCnt() > emptyCount {
```

Comment:
Not sure why do you have emptyCount variable. Anyways you are using only one place. instead , you can directly use n.hasLoadBalancerEndpoint().

Reply:
The code is attempting to determine whether there are any *container& endpoints left in the network.  If there are, then it can not delete the network.  The code does this by querying `n.getEpCnt().EndpointCnt()` to get the current number of endpoints connected to the network.  If the network has a load balancer endpoint, then a count of 1 indicates that there are no container endpoints in the network.  If there is no load balancer endpoint, then only a count of 0 indicates that there are no more container endpoints.